@here/harp-mapview/lib/Tile.ts

```diff
@@ -505,6 +505,10 @@ export class Tile implements CachedResource {
      * @param textElement The Text element to add.
      */
     addUserTextElement(textElement: TextElement) {
+        if (this.m_textElementsChanged === false) {
+            this.m_userTextElements = this.m_userTextElements.clone();
```

Comment:
The fix is mainly to recreate a text element group when it's changed, so that TextElementsRenderer knows it's a different group and doesn't try to match it with the old group state.
I'm creating another PR later to cleanup the Tile interface (we are gonna deprecate user text elements) and document all this.

Reply:
Crazy, please add a small comment that this will be cleaned up in a separate commit. Also, you are checking === false, i.e. it didn't change, is this wanted?

Reply:
done.
Yes, we want to clone the text element group once, just when we are about to change it since the last time it was rendered. 
After the first change (m_textElementsChanged === true), we don't need to clone it anymore.

Reply:
Now I understand, ok, cool.